app/controllers/listings_controller.rb

```diff
@@ -7,7 +7,7 @@ class ListingsController < ApplicationController
 
   def create
     @from = params[:from]
-    @movie = Movie.find_by(tmdb_id: params[:tmdb_id]) || tmdb_handler_add_movie(params[:tmdb_id])
+    @movie = GuaranteedMovie.find(params[:tmdb_id])
```

Comment:
The `tmdb_handler_add_movie` felt funny to me -- like it didn't belong in the `Tmdb::Client` space or the `Movie` modal space. Then when I saw it in the context of where it is being used, I decided that it could live in a `GuaranteedMovie` class that also handles the `||` logic.